### ISSUE TITLE
Configuration for rule "unicorn/explicit-length-check" is invalid

### ISSUE THREAD

**Issue by stheine** [sindresorhus/eslint-plugin-unicorn#1844]
1️⃣ Explain here what's wrong

```
Error: .eslintrc.cjs » eslint-config-es/node:
        Configuration for rule "unicorn/explicit-length-check" is invalid:
        Value {"nonZero":"greaterThan","non-zero":"greater-than"} should NOT have additional properties.
```

this is showing since `39.0.0`

2️⃣ Specify which rule is buggy here and in the title

unicorn/explicit-length-check

3️⃣ Add some examples where the issue appears

`package.json`:
```json
{
  "name": "test",
  "version": "1.0.0",
  "description": "",
  "type": "module",
  "main": "test.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "eslint": "8.18.0",
    "eslint-config-es": "4.2.0",
    "eslint-plugin-eslint-comments": "3.2.0",
    "eslint-plugin-extended": "0.2.0",
    "eslint-plugin-import": "2.26.0",
    "eslint-plugin-mocha": "10.0.5",
    "eslint-plugin-unicorn": "42.0.0"
  }
}
```
`.eslintrc.cjs`:
```js
module.exports = {
  extends: [
    'es/node'
  ],
  parserOptions: {
    sourceType: 'module'
  },
  rules: {
  }
}
```

```
eslint test.js
```

results in the error
```
Error: .eslintrc.cjs » eslint-config-es/node:
        Configuration for rule "unicorn/explicit-length-check" is invalid:
        Value {"nonZero":"greaterThan","non-zero":"greater-than"} should NOT have additional properties.
```

and it's not possible to even disable that rule by:

```js
  rules: {
    'unicorn/explicit-length-check': 0
  }
```

**Comment by fisker** [comment]
Here is the example https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/explicit-length-check.md#options
